lib/ansible/module_utils/vmware.py

```diff
@@ -281,6 +289,47 @@ def gather_vm_facts(content, vm):
         'current_snapshot': None,
     }
 
+    # facts that may or may not exist
+    if vm.summary.runtime.host:
+        host = vm.summary.runtime.host
+        facts['hw_esxi_host'] = host.summary.config.name
+
+    datastores = vm.datastore
+    for ds in datastores:
+        facts['hw_datastores'].append(ds.info.name)
+
+    try:
+        files = vm.config.files
+        layout = vm.layout
+        if files:
+            facts['hw_files'] = [files.vmPathName]
+            for item in layout.snapshot:
+                for snap in item.snapshotFile:
+                    facts['hw_files'].append(files.snapshotDirectory + snap)
+            for item in layout.configFile:
+                facts['hw_files'].append(os.path.dirname(files.vmPathName) + '/' + item)
+            for item in vm.layout.logFile:
+                facts['hw_files'].append(files.logDirectory + item)
+            for item in vm.layout.disk:
+                for disk in item.diskFile:
+                    facts['hw_files'].append(disk)
+    except:
+        pass
+
+    folder = vm.parent
```

Comment:
Can we use `compile_folder_path_for_object` here ?

Reply:
Apparently not.  It would appear that compile_folder_path_for_object gives a different result:
            "hw_folder": "/DC0/vm", 
            "hw_folderx": "/vm", 
(folderx is the compile_fpfo return)  Perhaps that function should be changed, but I don't know all the places it's actually used, and that might cause pain elsewhere.
I specifically want the full path including datacenter, so it can be directly used by other calls.